apiserver/logsink.go

```diff
@@ -61,24 +143,44 @@ func (s *agentLoggingStrategy) init(ctxt httpContext, req *http.Request) error {
 	// address this caveat appropriately.
 	ver, err := logsink.JujuClientVersionFromRequest(req)
 	if err != nil {
-		releaser()
+		releaseState()
 		return errors.Trace(err)
 	}
-	s.releaser = releaser
 	s.version = ver
 	s.entity = entity.Tag()
 	s.filePrefix = st.ModelUUID() + ":"
-	s.dbLogger = state.NewEntityDbLogger(st, s.entity, s.version)
+	s.dblogger = s.dbloggers.get(st)
+	s.releaser = func() {
+		if removed := releaseState(); removed {
```

Comment:
more that the *State object is no longer usable, but effectively yes; because we don't remove them until the model is dead